Network/Network.cpp

```diff
@@ -625,154 +654,277 @@ namespace WPEFramework
-                        broadcast_addr2.s_addr = gateway_address.s_addr | ~mask.s_addr;
-
-                        if (ip_address.s_addr == gateway_address.s_addr)
-                        {
-                            LOGWARN("Interface and Gateway IP are same , return false \n");
-                            response["supported"] = iarmData.isSupported;
-                            result = false;
-                            returnResponse(result)
-                        }
-                        if (broadcast_addr1.s_addr != broadcast_addr2.s_addr)
-                        {
-                            LOGWARN("Interface and Gateway IP is not in same broadcast domain, return false \n");
-                            response["supported"] = iarmData.isSupported;
-                            result = false;
-                            returnResponse(result)
-                        }
-                        if (ip_address.s_addr == broadcast_addr1.s_addr)
-                        {
-                            LOGWARN("Interface and Broadcast IP is same, return false \n");
-                            response["supported"] = iarmData.isSupported;
-                            result = false;
-                            returnResponse(result)
-                        }
-                        if (gateway_address.s_addr == broadcast_addr2.s_addr)
-                        {
-                            LOGWARN("Gateway and Broadcast IP is same, return false \n");
-                            response["supported"] = iarmData.isSupported;
-                            result = false;
-                            returnResponse(result)
-                        }
+                        LOGWARN("Interface and Broadcast IP is same, return false \n");
+                        response["supported"] = iarmData.isSupported;
+                        result = false;
+                        returnResponse(result)
+                    }
+                    if (gateway_address.s_addr == broadcast_addr2.s_addr)
+                    {
+                        LOGWARN("Gateway and Broadcast IP is same, return false \n");
+                        response["supported"] = iarmData.isSupported;
+                        result = false;
+                        returnResponse(result)
                     }
                 }
-                if (IARM_RESULT_SUCCESS ==
-                    IARM_Bus_Call(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETSRVMGR_API_setIPSettings, (void *) &iarmData,
-                                  sizeof(iarmData)))
-                {
-                    response["supported"] = iarmData.isSupported;
-                    result = true;
-                }
-                else
-                    response["supported"] = iarmData.isSupported;
             }
+            else
+            {
+                response["supported"] = iarmData.isSupported;
+                result = true;
+                returnResponse(result)
```

Comment:
this means, you never call netsrvmgr to set the autoconfig IP.
The ```returnResponse(result)``` is nothing but ```return SUCCESS;``` The below code will never get executed.